src/ssh/azext_ssh/tests/latest/test_resource_tag_utils.py

```diff
@@ -0,0 +1,122 @@
+import unittest
+from unittest import mock
+from azure.cli.core import azclierror
+from azext_ssh import resource_tag_utils
+from azext_ssh.ssh_info import SSHSession
+
+class TestHandleResourceTagConfig(unittest.TestCase):
+    def setUp(self):
+        self.cmd = mock.Mock()
+        self.op_info = SSHSession("rg", "vm", None, None, None, False, None, None, None, None, [], False, "Microsoft.Compute/virtualMachines", None, None, False, False, None)
+
+    # Tests for scenarios with no port and no resource tag (Default Run)
+    @mock.patch('azext_ssh.resource_tag_utils._check_azure_vm_tag')
```

Comment:
I'm glad you wrote a lot of tests for this feature, but just to save you time in the future, ask for a code design review before writing tests. Because if you write a lot of tests, and then during the review you get comments to make a lot of changes you will have to re-write the tests.

Reply:
Noted thanks for the heads up